webpush/push.go

```diff
@@ -69,6 +69,42 @@ func NewPushRequest(sub *Subscription, message string, token string) (*http.Requ
 	return req, nil
 }
 
+// NewVapidRequest creates a valid Web Push HTTP request for sending a message
+// to a subscriber, using Vapid authentication. You can add more headers to
+// configure collapsing, TTL.
+func NewVapidRequest(sub *Subscription, message string, vapid *Vapid) (*http.Request, error) {
```

Comment:
I would rather remove NewPushRequest, and rename this method.

NewPushRequest uses a token, which is not in the standard - I can add a 'legacyToken' to vapid class, or as a config somewhere - to allow use with old tokens - but I would rather not have it in the main interface.